File: android/src/oldarch/com/appodeal/rnappodeal/RNAppodealNativeAssetViewManager.java
package com.appodeal.rnappodeal;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

public class RNAppodealNativeAssetViewManager extends SimpleViewManager<RCTAppodealNativeAssetView> {

    private final RNAppodealNativeAssetViewManagerImpl impl = new RNAppodealNativeAssetViewManagerImpl();

    @NonNull
    @Override
    public String getName() {
        return RNAppodealNativeAssetViewManagerImpl.NAME;
    }

    @NonNull
    @Override
    protected RCTAppodealNativeAssetView createViewInstance(@NonNull ThemedReactContext reactContext) {
        return impl.createViewInstance(reactContext);
    }

    @Override
    public void onDropViewInstance(@NonNull RCTAppodealNativeAssetView view) {
        super.onDropViewInstance(view);
        impl.onDropViewInstance(view);
    }

    @ReactProp(name = "asset")
    public void setAsset(RCTAppodealNativeAssetView view, @Nullable String value) {
        if (value != null) {
            view.setAssetType(value);
        }
    }
}

File: android/src/newarch/com/appodeal/rnappodeal/RNAppodealNativeAssetViewManager.kt
package com.appodeal.rnappodeal

import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

@ReactModule(name = RNAppodealNativeAssetViewManagerImpl.NAME)
class RNAppodealNativeAssetViewManager :
    SimpleViewManager<RCTAppodealNativeAssetView>() {

    private val impl = RNAppodealNativeAssetViewManagerImpl()

    override fun getName(): String = RNAppodealNativeAssetViewManagerImpl.NAME

    override fun createViewInstance(reactContext: ThemedReactContext): RCTAppodealNativeAssetView {
        return impl.createViewInstance(reactContext)
    }

    override fun onDropViewInstance(view: RCTAppodealNativeAssetView) {
        super.onDropViewInstance(view)
        impl.onDropViewInstance(view)
    }

    @ReactProp(name = "asset")
    fun setAsset(view: RCTAppodealNativeAssetView, value: String?) {
        value?.let { view.assetType = it }
    }
}
